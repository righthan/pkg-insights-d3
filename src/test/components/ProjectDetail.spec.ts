import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils' 
import ProjectDetailVue from '@/components/ProjectDetail.vue';

describe('PkgDetail test', () => {
    const data = {
        entryPackageName: '包名',
        entryVersion: '版本',
        depth: 5,
        isCircle: true,
        isMulPackage: true,
        nodeCount: 10,
        circleDepList: [['A', 'B', 'C'], ['D', 'E', 'F']],
        mulPackageList: [['链接1', '链接2'], ['链接3', '链接4']],
    };
    test('组件渲染测试', () => {
        const wrapper = mount(ProjectDetailVue, {
            props: {
                data,
                isLocalFile:false
            }
        });
        expect(wrapper.html()).toMatchSnapshot();
    })
    test('点击链接时正确触发handleClickLink方法并触发refresh事件', async () => {
    
        const wrapper = mount(ProjectDetailVue, {
            props: {
                data,
                isLocalFile:false
            },
        });
    
        // 模拟点击多版本依赖链接
        const mulPackageLink = wrapper.find('.links-box .el-link');
        await mulPackageLink.trigger('click');
    
        // 验证handleClickLink方法是否被正确触发
        expect(wrapper.emitted('searchNode')).toBeTruthy();
    
        // 验证searchNode事件是否传递了正确的参数
        expect((wrapper.emitted('searchNode') as any)[0][0]).toBe('链接1');
    
        // 模拟点击循环依赖链接
        const circleDepLink = wrapper.find('.links .el-link');
        await circleDepLink.trigger('click');
    
        // 验证handleClickLink方法是否被正确触发
        expect(wrapper.emitted('searchNode')).toBeTruthy();
    
        // 验证searchNode事件是否传递了正确的参数
        expect((wrapper.emitted('searchNode') as any)[1][0]).toBe('A');
    });
    test('点击"查看"链接时正确触发showMulPackage改变', async () => {
        const wrapper = mount(ProjectDetailVue, {
            props: {
                data,
                isLocalFile:false
            },
        });
        const vm = wrapper.vm as any;
        const flag = vm.showMulPackage
        // 模拟点击"查看"链接
        const viewMulPackageLink = wrapper.find('.el-link');
        await viewMulPackageLink.trigger('click')
        expect(vm.showMulPackage).toBe(!flag);
    });
    test('点击"查看"链接时正确触发showCirleDep改变', async () => {
        const wrapper = mount(ProjectDetailVue, {
            props: {
                data,
                isLocalFile:false
            },
        });
        const vm = wrapper.vm as any;
        const flag = vm.showCirleDep
        // 模拟点击"查看"链接
        const viewMulPackageLink = wrapper.findAll('.desc-link .el-link')[0];
        
        await viewMulPackageLink.trigger('click')
        expect(vm.showCirleDep).toBe(!flag);
    });
    test('点击"图像"链接时正确触发hilightCirleLinks事件', async () => {
        const data = {
          entryPackageName: '包名',
          entryVersion: '版本',
          depth: 5,
          isCircle: true,
          isMulPackage: true,
          nodeCount: 10,
          circleDepList: [['A', 'B', 'C'], ['D', 'E', 'F']],
          mulPackageList: [['链接1', '链接2'], ['链接3', '链接4']],
        };
    
        const wrapper = mount(ProjectDetailVue, {
          props: {
            data,
            isLocalFile: false,
          },
        });
        
        // 模拟点击"图像"链接
        const hightCirleLinksLink = wrapper.findAll('.desc-link .el-link')[1]
        
        await hightCirleLinksLink.trigger('click');
    
        // 验证handleHightCirleLinks事件是否被正确触发
        expect(wrapper.emitted('hilightCirleLinks')).toBeTruthy();
      });
})