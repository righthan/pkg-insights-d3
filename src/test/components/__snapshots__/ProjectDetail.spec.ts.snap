// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PkgDetail test > 组件渲染测试 1`] = `
"<div data-v-460d2e40=\\"\\" class=\\"el-descriptions\\">
  <div class=\\"el-descriptions__header\\">
    <div class=\\"el-descriptions__title\\">根项目详情</div>
    <div class=\\"el-descriptions__extra\\"></div>
  </div>
  <div class=\\"el-descriptions__body\\">
    <table class=\\"el-descriptions__table is-bordered\\">
      <tbody>
        <tr>
          <td style=\\"min-width: 150px;\\" class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">项目名称</td>
          <td style=\\"min-width: 150px;\\" class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">包名</td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">版本</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">版本</td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">分析依赖层数</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">
            <div data-v-460d2e40=\\"\\" class=\\"el-select el-select--small m-2\\">
              <div class=\\"select-trigger el-tooltip__trigger el-tooltip__trigger\\">
                <!--v-if-->
                <!-- fix: https://github.com/element-plus/element-plus/issues/11415 -->
                <!--v-if-->
                <div class=\\"el-input el-input--small el-input--suffix\\">
                  <!-- input -->
                  <!-- prepend slot -->
                  <!--v-if-->
                  <div class=\\"el-input__wrapper\\">
                    <!-- prefix slot -->
                    <!--v-if--><input class=\\"el-input__inner\\" type=\\"text\\" readonly=\\"\\" autocomplete=\\"off\\" tabindex=\\"0\\" placeholder=\\"Select\\"><!-- suffix slot --><span class=\\"el-input__suffix\\"><span class=\\"el-input__suffix-inner\\"><i class=\\"el-icon el-select__caret el-select__icon\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 1024 1024\\"><path fill=\\"currentColor\\" d=\\"M831.872 340.864 512 652.672 192.128 340.864a30.592 30.592 0 0 0-42.752 0 29.12 29.12 0 0 0 0 41.6L489.664 714.24a32 32 0 0 0 44.672 0l340.288-331.712a29.12 29.12 0 0 0 0-41.728 30.592 30.592 0 0 0-42.752 0z\\"></path></svg></i><!--v-if--><!--v-if--><!--v-if--><!--v-if--><!--v-if--><!--v-if--></span></span>
                  </div><!-- append slot -->
                  <!--v-if-->
                </div>
              </div>
              <!--teleport start-->
              <!--teleport end-->
            </div>
          </td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">分析节点数</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">10</td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">是否有多版本依赖</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">
            <transition-stub data-v-460d2e40=\\"\\" name=\\"el-zoom-in-center\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><span class=\\"el-tag el-tag--danger el-tag--light\\"><span class=\\"el-tag__content\\">是</span>
              <!--v-if--></span>
            </transition-stub><span data-v-460d2e40=\\"\\">   <a data-v-460d2e40=\\"\\" class=\\"el-link el-link--success\\"><!--v-if--><span class=\\"el-link__inner\\">查看</span>
            <!--v-if--></a></span>
          </td>
        </tr>
        <tr>
          <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">是否有循环依赖</td>
          <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"1\\">
            <transition-stub data-v-460d2e40=\\"\\" name=\\"el-zoom-in-center\\" appear=\\"true\\" persisted=\\"false\\" css=\\"true\\"><span class=\\"el-tag el-tag--danger el-tag--light\\"><span class=\\"el-tag__content\\">是</span>
              <!--v-if--></span>
            </transition-stub><span data-v-460d2e40=\\"\\" class=\\"desc-link\\">   <a data-v-460d2e40=\\"\\" class=\\"el-link el-link--success\\"><!--v-if--><span class=\\"el-link__inner\\">查看</span>
            <!--v-if--></a>     <a data-v-460d2e40=\\"\\" class=\\"el-link el-link--success\\">
              <!--v-if--><span class=\\"el-link__inner\\">图像</span>
              <!--v-if-->
            </a></span>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
<div data-v-460d2e40=\\"\\" style=\\"display: none;\\">
  <h4 data-v-460d2e40=\\"\\" class=\\"desc-title\\">项目多版本依赖详情</h4>
  <div data-v-460d2e40=\\"\\" class=\\"el-scrollbar\\">
    <div class=\\"el-scrollbar__wrap el-scrollbar__wrap--hidden-default\\" style=\\"max-height: 150px;\\">
      <div class=\\"el-scrollbar__view\\" style=\\"\\">
        <div data-v-460d2e40=\\"\\" class=\\"el-descriptions\\">
          <!--v-if-->
          <div class=\\"el-descriptions__body\\">
            <table class=\\"el-descriptions__table is-bordered\\">
              <tbody>
                <tr>
                  <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">链接1</td>
                  <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"NaN\\">
                    <div data-v-460d2e40=\\"\\" class=\\"links-box\\"><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                        <!--v-if--><span class=\\"el-link__inner\\"></span>
                        <!--v-if-->
                      </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                        <!--v-if--><span class=\\"el-link__inner\\"></span>
                        <!--v-if-->
                      </a></div>
                  </td>
                </tr>
                <tr>
                  <td class=\\"el-descriptions__cell el-descriptions__label is-bordered-label\\" colspan=\\"1\\">链接3</td>
                  <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content\\" colspan=\\"1\\">
                    <div data-v-460d2e40=\\"\\" class=\\"links-box\\"><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                        <!--v-if--><span class=\\"el-link__inner\\"></span>
                        <!--v-if-->
                      </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                        <!--v-if--><span class=\\"el-link__inner\\"></span>
                        <!--v-if-->
                      </a></div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <transition-stub name=\\"el-scrollbar-fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
      <div class=\\"el-scrollbar__bar is-horizontal\\" style=\\"display: none;\\">
        <div class=\\"el-scrollbar__thumb\\" style=\\"width: 0px; transform: translateX(0%);\\"></div>
      </div>
    </transition-stub>
    <transition-stub name=\\"el-scrollbar-fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
      <div class=\\"el-scrollbar__bar is-vertical\\" style=\\"display: none;\\">
        <div class=\\"el-scrollbar__thumb\\" style=\\"height: 0px; transform: translateY(0%);\\"></div>
      </div>
    </transition-stub>
  </div>
</div>
<div data-v-460d2e40=\\"\\" style=\\"display: none;\\">
  <div data-v-460d2e40=\\"\\" class=\\"el-descriptions\\">
    <!--v-if-->
    <div class=\\"el-descriptions__body\\">
      <table class=\\"el-descriptions__table is-bordered\\">
        <tbody>
          <tr>
            <th class=\\"el-descriptions__cell el-descriptions__label is-bordered-label is-vertical-label\\" colspan=\\"1\\">循环依赖详情</th>
          </tr>
          <tr>
            <td class=\\"el-descriptions__cell el-descriptions__content is-bordered-content is-vertical-content\\" colspan=\\"1\\">
              <div data-v-460d2e40=\\"\\" class=\\"el-scrollbar\\">
                <div class=\\"el-scrollbar__wrap el-scrollbar__wrap--hidden-default\\" style=\\"max-height: 150px;\\">
                  <div class=\\"el-scrollbar__view\\" style=\\"\\">
                    <div data-v-460d2e40=\\"\\">
                      <div data-v-460d2e40=\\"\\" class=\\"links\\"><span data-v-460d2e40=\\"\\">[</span><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                          <!--v-if--><span class=\\"el-link__inner\\">A</span>
                          <!--v-if-->
                        </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary arrow\\">
                          <!--v-if--><span class=\\"el-link__inner\\">B</span>
                          <!--v-if-->
                        </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary arrow\\">
                          <!--v-if--><span class=\\"el-link__inner\\">C</span>
                          <!--v-if-->
                        </a><span data-v-460d2e40=\\"\\">]</span></div>
                    </div>
                    <div data-v-460d2e40=\\"\\">
                      <div data-v-460d2e40=\\"\\" class=\\"links\\"><span data-v-460d2e40=\\"\\">[</span><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary\\">
                          <!--v-if--><span class=\\"el-link__inner\\">D</span>
                          <!--v-if-->
                        </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary arrow\\">
                          <!--v-if--><span class=\\"el-link__inner\\">E</span>
                          <!--v-if-->
                        </a><a data-v-460d2e40=\\"\\" class=\\"el-link el-link--primary arrow\\">
                          <!--v-if--><span class=\\"el-link__inner\\">F</span>
                          <!--v-if-->
                        </a><span data-v-460d2e40=\\"\\">]</span></div>
                    </div>
                  </div>
                </div>
                <transition-stub name=\\"el-scrollbar-fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
                  <div class=\\"el-scrollbar__bar is-horizontal\\" style=\\"display: none;\\">
                    <div class=\\"el-scrollbar__thumb\\" style=\\"width: 0px; transform: translateX(0%);\\"></div>
                  </div>
                </transition-stub>
                <transition-stub name=\\"el-scrollbar-fade\\" appear=\\"false\\" persisted=\\"true\\" css=\\"true\\">
                  <div class=\\"el-scrollbar__bar is-vertical\\" style=\\"display: none;\\">
                    <div class=\\"el-scrollbar__thumb\\" style=\\"height: 0px; transform: translateY(0%);\\"></div>
                  </div>
                </transition-stub>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>"
`;
